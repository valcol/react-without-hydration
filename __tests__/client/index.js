/**
 * @jest-environment jsdom
 */

import React from "react";
import ReactDom from "react-dom";
import withoutHydration from "../../src";

const Component = ({ label }) => <div className="label">{label}</div>;
const SSRhtml =
    '<section data-no-hydrate="true"><div class="label">some content sever side</div></section>';

describe("With SSR", () => {
    test("Render correctly client side, no option ", () => {
        const elem = document.createElement("div");
        elem.innerHTML = SSRhtml;

        const ComponentwithoutHydrationClient = withoutHydration()(Component);

        ReactDom.hydrate(
            <ComponentwithoutHydrationClient label="some content client side" />,
            elem
        );

        expect(elem).toMatchSnapshot();
    });

    test("Render correctly client side, onUpdate ", () => {
        const elem = document.createElement("div");
        elem.innerHTML = SSRhtml;
        const onUpdate = jest.fn(({ label }, domNode) => {
            domNode.getElementsByClassName("label")[0].innerHTML = label;
        });

        const ComponentwithoutHydrationClient = withoutHydration({
            onUpdate
        })(Component);

        ReactDom.hydrate(
            <ComponentwithoutHydrationClient label="some content set client side via onUpdate" />,
            elem
        );

        expect(onUpdate).toHaveBeenCalled();
        expect(elem).toMatchSnapshot();
    });

    test("Render correctly client side, forceHydration ", () => {
        const elem = document.createElement("div");
        elem.innerHTML = SSRhtml;

        const ComponentwithoutHydrationClient = withoutHydration()(Component);

        ReactDom.hydrate(
            <ComponentwithoutHydrationClient
                label="some content client side"
                forceHydration
            />,
            elem
        );

        expect(elem).toMatchSnapshot();
    });
});

describe("Without SSR", () => {
    test("Render correctly client side, no option ", () => {
        const elem = document.createElement("div");

        const ComponentwithoutHydrationClient = withoutHydration()(Component);

        ReactDom.hydrate(
            <ComponentwithoutHydrationClient label="some content client side" />,
            elem
        );

        expect(elem).toMatchSnapshot();
    });

    test("Render correctly client side, onUpdate ", () => {
        const elem = document.createElement("div");
        const onUpdate = jest.fn();

        const ComponentwithoutHydrationClient = withoutHydration({
            onUpdate
        })(Component);

        ReactDom.hydrate(
            <ComponentwithoutHydrationClient label="some content client side" />,
            elem
        );

        expect(onUpdate).not.toHaveBeenCalled();
        expect(elem).toMatchSnapshot();
    });

    test("Render correctly client side, disableFallback at true", () => {
        const elem = document.createElement("div");

        const ComponentwithoutHydrationClient = withoutHydration({
            disableFallback: true
        })(Component);

        ReactDom.hydrate(
            <ComponentwithoutHydrationClient label="some content client side" />,
            elem
        );

        expect(elem).toMatchSnapshot();
    });
});
