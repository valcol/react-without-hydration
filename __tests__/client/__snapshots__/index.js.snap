// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With SSR Render correctly client side, forceHydration  1`] = `
<div>
  <div
    class="label"
  >
    some content client side
  </div>
</div>
`;

exports[`With SSR Render correctly client side, no option  1`] = `
<div>
  <section
    data-no-hydrate="true"
  >
    <div
      class="label"
    >
      some content sever side
    </div>
  </section>
</div>
`;

exports[`With SSR Render correctly client side, onUpdate  1`] = `
<div>
  <section
    data-no-hydrate="true"
  >
    <div
      class="label"
    >
      some content set client side via onUpdate
    </div>
  </section>
</div>
`;

exports[`Without SSR Render correctly client side, disableFallback at true 1`] = `
<div>
  <section />
</div>
`;

exports[`Without SSR Render correctly client side, no option  1`] = `
<div>
  <div
    class="label"
  >
    some content client side
  </div>
</div>
`;

exports[`Without SSR Render correctly client side, onUpdate  1`] = `
<div>
  <div
    class="label"
  >
    some content client side
  </div>
</div>
`;
